src/content/getusermedia/getdisplaymedia/js/main.js

```diff
@@ -6,40 +6,127 @@
+          fromAttribute: input => input
+        }
+      }
+    };
+  }
 
-  // demonstrates how to detect that the user has stopped
-  // sharing the screen via the browser UI.
-  stream.getVideoTracks()[0].addEventListener('ended', () => {
-    errorMsg('The user has ended sharing the screen');
-  });
+  render() {
+    return html`<style>
+@import "../../../css/main.css";
+:host {
+  display: block;
+  padding: 10px;
+  width: 100%;
+  height: 100%;
 }
-
-function handleError(error) {
-  errorMsg(`getDisplayMedia error: ${error.name}`, error);
+video {
+    --width: 100%;
+    width: var(--width);
+    height: calc(var(--width) * (16 / 9));
 }
 
-function errorMsg(msg, error) {
-  const errorElement = document.querySelector('#errorMsg');
-  errorElement.innerHTML += `<p>${msg}</p>`;
-  if (typeof error !== 'undefined') {
-    console.error(error);
+</style>
+<video playsinline autoplay muted .srcObject="${this.stream}"></video>
+<div>
+<button ?disabled="${!this.buttons.startCapturing}" @click="${async e => this._startCapturing(e)}">Start screen capture</button>
+<button ?disabled="${!this.buttons.stopCapturing}" @click="${e => this._stopCapturing(e)}">Stop screen capture</button>
+<button ?disabled="${!this.buttons.downloadRecording}" @click="${e => this._downloadRecording(e)}">Download recording</button>
+<a id="downloadLink" type="video/webm" style="display: none"></a>
+</div>`;
   }
-}
 
-if ('getDisplayMedia' in navigator) {
-  navigator.getDisplayMedia({video: true})
-    .then(handleSuccess)
-    .catch(handleError);
-} else {
-  errorMsg('getDisplayMedia is not supported');
+  _startScreenCapture() {
+    if (navigator.getUserMedia) {
+      return navigator.getDisplayMedia({video: true});
+    } else {
+      return navigator.mediaDevices.getUserMedia({video: {mediaSource: 'screen'}});
+    }
+  }
+
+  async _startCapturing(e) {
+    console.log('Start capturing.');
+    this.buttons.startCapturing = false;
+    this.buttons.stopCapturing = true;
+    this.buttons.downloadRecording = false;
+    this.requestUpdate('buttons');
+
+    if (this.downloadUrl) {
+      window.URL.revokeObjectURL(this.downloadUrl);
+    }
+
+    this.recording = [];
+    this.stream = await this._startScreenCapture();
+    this.stream.addEventListener('inactive', e => {
+      console.log('Capture stream inactive - stop recording!');
+      this.buttons.startCapturing = true;
+      this.buttons.stopCapturing = false;
+      this.buttons.downloadRecording = true;
+      this.requestUpdate('buttons');
+
+      this.mediaRecorder.stop();
+      this.mediaRecorder = null;
+      this.stream.getTracks().forEach(track => track.stop());
+      this.stream = null;
+    });
+    this.mediaRecorder = new MediaRecorder(this.stream, {mimeType: 'video/webm'});
+    this.mediaRecorder.addEventListener('dataavailable', event => {
+      if (event.data && event.data.size > 0) {
+        this.recording.push(event.data);
+      }
+    });
+    this.mediaRecorder.start(10);
```

Comment:
What's the rationale behind a 10ms timeslice?

Your use of the timeslice, namely storing the blobs in memory, is functionally equivalent to not having a timeslice at all.

Not having a timeslice also adds some benefits. It lets the UA optimize how it gathers data into the blobs, e.g., by storing the data in a blob on disk instead of in memory. It also simplifies your code since you don't need the `recording` array and the bookkeeping that comes with it.

Reply:
That part is a copy from another sample. Since this sample is going away (from the massive negative feedback I got so far) I won't update this for now, but please have a look at the MediaRecorder sample and suggest an improvement there. Thanks!

Reply:
Gotcha. Filed issue #1153.